libats-messaging-datatype/src/main/scala/com/advancedtelematic/libats/messaging_datatype/Messages.scala

```diff
@@ -168,6 +169,10 @@ object Messages {
       rawReport: Option[String] = None
   ) extends DeviceUpdateEvent
 
+  final case class SystemInfo(product: Option[String])
+
+  final case class DeviceSystemInfoChanged(namespace: Namespace, uuid: DeviceId, newSystemInfo: Option[SystemInfo])
```

Comment:
So `newSystemInfo` is an `Option` within an `Option`? Couldn't we have `product: String`?

Reply:
No, because we want to add more things other than product, and the message is still valid without any system info at all. In practice right now it if it doesn't have a product it also doesn't have a system info, but with more attributes it could not have a product but still have a system info.

Also, makes the migration easier because old messages can still be parsed, they just have None on the newSystemInfo field.

Reply:
Alright, makes sense 👍